src/app/components/cohort-charts/chart.service.js

```diff
@@ -6,8 +6,11 @@ angular.module('transmartBaseUi').factory('ChartService',
 
             var chartService = {
                 cs: {},
-                nodes: [],
-                cohortUpdating: false
+                cohortUpdating: false,
+                // the filters store a map --
+                // key: chart.tsLabel.label (or node.restObj.fullName, the two are the same),
+                // val: an array of filters of that chart
+                filters: {}
```

Comment:
I think we dont have to store filters since we can get the current filter on each chart by _chart.filter()